slides/k8s/configuration.md

```diff
@@ -0,0 +1,529 @@
+  ```
+
+- Check what our configmap looks like:
+  ```bash
+  kuebectl get configmap haproxy -o yaml
+  ```
+
+]
+
+---
+
+## Using the configmap
+
+We are going to use the following pod definition:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: haproxy
+spec:
+  volumes:
+  - name: config
+    configMap:
+      name: haproxy
+  containers:
+  - name: haproxy
+    image: haproxy
+    volumeMounts:
+    - name: config
+      mountPath: /usr/local/etc/haproxy/
+```
+
+---
+
+## Using the configmap
+
+- The resource definition from the previous slide is in `k8s/haproxy.yaml`
+
+.exercise[
+
+- Create the HAProxy pod:
+  ```bash
+  kubectl apply -f ~/container.training/k8s/haproxy.yaml
+  ```
+
+- Check the IP address allocated to the pod:
+  ```bash
+  kubectl get pod haproxy -o wide
+  IP=$(kubectl get pod haproxy -o json | jq -r .status.podIP)
+  ```
+
+]
+
+---
+
+## Testing our load balancer
+
+- The load balancer will send:
+
+  - half of the connections to Google
+
+  - the other half to Bing
+
+.exercise[
+
+- Access the load balancer a few times:
+  ```bash
+  curl -I $IP
+  curl -I $IP
+  curl -I $IP
+  ```
+
+]
+
+We should see connections served by Google (look for the `Location` header) and others served by Bing (indicated by the `X-MSEdge-Ref` header).
+
+---
+
+## Exposing configmaps with the downward API
+
+- We are going to run a Docker registry on a custom port
+
+- By default, the registry listens on port 5000
+
+- This can be changed by setting environment variable `REGISTRY_HTTP_ADDR`
+
+- We are going to store the port number in a configmap
+
+- Then we will expose that configmap to a container environment variable
+
+---
+
+## Creating the configmap
+
+.exercise[
+
+- Our configmap will have a single key, `http.addr`:
+  ```bash
+  kubectl create configmap registry --from-literal=http.addr=0.0.0.0:80
+  ```
+
+- Check our configmap:
+  ```bash
+  kubectl get configmap regsitry -o yaml
+  ```
+
+]
+
+---
+
+## Using the configmap
+
+We are going to use the following pod definition:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: registry
+spec:
+  containers:
+  - name: registry
+    image: registry
+    env:
+    - name: REGISTRY_HTTP_ADDR
+      valueFrom:
+        configMapKeyRef:
+          name: registry
+          key: http.addr
+```
+
+---
+
+## Using the configmap
+
+- The resource definition from the previous slide is in `k8s/registry.yaml`
+
+.exercise[
+
+- Create the registry pod:
+  ```bash
+  kubectl apply -f ~/container.training/k8s/registry.yaml
+  ```
+
+- Check the IP address allocated to the pod:
+  ```bash
+  kubectl get pod registry -o wide
+  IP=$(kubectl get pod registry -o json | jq -r .status.podIP)
+  ```
+
+- Confirm that the registry is available on port 80:
+  ```bash
+  curl $IP/v2/_catalog
+  ```
+
+]
+
+---
+
+## Passwords, tokens, sensitive information
+
+- For sensitive information, there is another special resource: *Secrets*
```

Comment:
I figured this was coming, but when you start talking about where to put variables, maybe foreshadow this? Then people won't be sitting there thinking "BUT WAIT, WHAT ABOUT SECRETS"?

Reply:
I was wondering if I should list it on the slide "Passing configuration to containers", but technically secrets and configmaps are not different (in the way the information is passed to the containers) so I thought it would add confusion ... I'll see if I can find a way to tip our hand without confusing things further!

Reply:
Maybe just mention that configmaps and secrets behave the same (and say yes! this is foreshadowing!) ?

Reply:
In the beginning of the configmap section, I'm adding this:

*Note: to hold sensitive information, se can use "Secrets", which are another type of resource behaving very much like configmaps. We'll cover them just after!*

Reply:
Good deal! I fixed the typo (changed `se` to `we`) and this is good to go.